readthedocs/search/tasks.py

```diff
@@ -142,19 +142,45 @@ def delete_old_search_queries_from_db():
 
 
 @app.task(queue='web')
-def record_search_query(project_slug, version_slug, query, total_results):
-    """Record search query in database."""
-    if not project_slug or not version_slug or not query or not total_results:
+def record_search_query(project_slug, version_slug, query, total_results, time):
+    """Record/update search query in database."""
+    if not project_slug or not version_slug or not query:
         log.debug(
             'Not recording the search query. Passed arguments: '
-            'project_slug: %s, version_slug: %s, query: %s, total_results: %s' % (
-                project_slug, version_slug, query, total_results
+            'project_slug: %s, version_slug: %s, query: %s, total_results: %s, time: %s' % (
+                project_slug, version_slug, query, total_results, time
             )
         )
         return
 
-    project_qs = Project.objects.filter(slug=project_slug)
+    before_10_sec = time - timezone.timedelta(seconds=10)
+    partial_query_qs = SearchQuery.objects.filter(
+        project__slug=project_slug,
+        version__slug=version_slug,
+        query__startswith=query,
+        created__gte=before_10_sec,
+    ).order_by('-created')
```

Comment:
I wonder about the performance characteristics of this. I don't believe `created` is indexed although `project` and `version` are. This might be worth testing on the live system before merging. This might be ok but I'm not 100% confident without just timing a similar query in prod.